protobuf-codegen-identical-test/tests/diff.rs

```diff
@@ -163,6 +267,22 @@ where
                 Path::new(&protoc_rs).strip_prefix(temp_dir.path()).unwrap(),
                 Path::new(&pure_rs).strip_prefix(temp_dir.path()).unwrap(),
             );
+
+            fs::write(
+                format!("{}/{}.descriptors", protoc_dir, proto_name),
+                pretty_message(&protoc_descriptors),
+            )
+            .unwrap();
+            fs::write(
+                format!("{}/{}.descriptors", pure_dir, proto_name),
+                pretty_message(&pure_descriptors),
+            )
+            .unwrap();
+            print_diff(
+                temp_dir.path(),
+                Path::new(&format!("protoc/{}.descriptors", proto_name)),
+                Path::new(&format!("pure/{}.descriptors", proto_name)),
+            );
```

Comment:
If you can, split this function so it could fit on one page.

Reply:
I didn't see a great way of splitting this up without passing a whole bunch of parameters around. Happy to try harder if you like.